gh-83405: Document None support for HTTPConnection.timeout (#17843)
python/cpython: Doc/library/http.client.rst

   string if it has the form ``host:port``, else the default HTTP port (80) is
   used.  If the optional *timeout* parameter is given, blocking

>    operations (like connection attempts) will timeout after that many seconds

Reviewer: ```suggestion
   operations (like connection attempts) will timeout after that many seconds.
```

>    If it is ``None`` then blocking operations will not timeout. If it is not given,

Reviewer 2: ```suggestion
   If it is ``None`` then blocking operations will not time out. If it is not given,
```
Spelling

   number.  If no port number is passed, the port is extracted from the host
   string if it has the form ``host:port``, else the default HTTP port (80) is
   used.  If the optional *timeout* parameter is given, blocking

>    operations (like connection attempts) will timeout after that many seconds.
>    If it is ``None`` then blocking operations will not time out. If it is not given,
>    the global default timeout setting is used. The optional *source_address*
>    parameter may be a tuple of a (host, port) to use as the source address the
>    HTTP connection is made from. The optional *blocksize* parameter sets the buffer
>    size in bytes for sending a file-like message body.

Reviewer: I'm personally no expert on the module so I can't quite give a proper review, but here's a few more comments:

Saying that the default timeout is used may give rise to the question of what it is, so if the following is accurate what do you think about including something like '... is used, which can be set by socket.settimeout()', since the actual variable being used (`socket._GLOBAL_DEFAULT_TIMEOUT`) isn't exposed?

On the side of formatting, to ease reviewing the changes, I would keep the original paragraph structure as close as possible, such as

```suggestion
   operations (like connection attempts) will timeout after that many seconds.
   If it is ``None`` then blocking operations will not time out. If it is not given,
   the global default timeout setting is used.
   The optional *source_address* parameter may be a tuple of a (host, port)
   to use as the source address the HTTP connection is made from.
   The optional *blocksize* parameter sets the buffer size in bytes for
   sending a file-like message body.
```

If I've done it right, only the addition should be shown after committing the change, which highlights the changes better. I wouldn't worry too much about how the raw paragraph looks as (imo) it's better to save review time than have a perfectly formatted rst file.

Another note is that there's [this other PR](https://github.com/python/cpython/pull/27087) introducing a similar note, which may somewhat invalidate these changes if it gets merged first. I do prefer this version a bit more as it integrates well into the paragraph, but I'll cross-reference this PR just as a heads up.

Reviewer 3: Perhaps you meant the _setdefaulttimeout_ module function, rather than _settimeout_ (a method)?
